### PR TITLE
replace sprintf with its safe version snprintf

### PR DESCRIPTION
**Pull request by hliu18**
sprintf is unsafe and has buffer overrun risk. Replace it with the safer version snprintf where buffer size is supplied to avoid overrun.

### CODE REVIEW

**Comment by siying** [comment]
CC @zhichao-cao

File: thirdparty.inc [hunk 1]
```diff
@@ -241,7 +241,7 @@ endif()
 
 if (WITH_JEMALLOC)
   message(STATUS "JEMALLOC library is enabled")
-  set(JEMALLOC_CXX_FLAGS "-DROCKSDB_JEMALLOC -DJEMALLOC_EXPORT= ")
+  set(JEMALLOC_CXX_FLAGS "-DROCKSDB_JEMALLOC -DJEMALLOC_EXPORT= -DJEMALLOC_NO_RENAME")
```
Comment (sagar0) [inline thirdparty.inc]: Why is this change in this PR? Assuming it got in here accidentally.

Comment (hliu18) [reply thirdparty.inc]: Yes, this change should not be in this PR.

Comment (sagar0) [reply thirdparty.inc]: Lets address the Jemalloc issue in #5470 .

**Comment by zhichao-cao** [comment]
lgtm, thanks.

**Pull request closed without merging**